tests/integration/questionnaire/test_questionnaire_calculated_summary.py

```diff
@@ -186,3 +186,94 @@ def test_calculated_summary_value_sources_across_sections_repeating(self):
+
+        self.post(
+            {
+                f"cost-of-transport-{list_item_ids[0]}": "100",
+                f"cost-of-transport-{list_item_ids[1]}": "200",
+            }
+        )
+        self.assertInBody(
+            "We calculate the total monthly spending on public transport to be £300.00. Is this correct?"
+        )
+        self.post()
+        self.post()
+        self.post()
+
+    def test_calculated_summary_repeating_and_static_answers(self):
+        """
+        This tests a calculated summary which uses both repeating and static answers
+        it pipes the calculated summary into a second section and disallows an answer greater than the total sum
+        """
+        self.launchSurvey("test_new_calculated_summary_repeating_and_static_answers")
+        self.post()
+        self.post({"any-supermarket-answer": "Yes"})
+        self.post({"supermarket-name": "Tesco"})
+        self.post({"list-collector-answer": "Yes"})
+        self.post({"supermarket-name": "Sainsburys"})
+        self.post({"list-collector-answer": "Yes"})
+        self.post({"supermarket-name": "Lidl"})
+
+        list_item_ids = self.get_list_item_ids()
+        assert len(list_item_ids) == 3
+        self.post({"list-collector-answer": "No"})
+
+        self.post(
+            {
+                f"cost-of-shopping-{list_item_ids[0]}": "100",
+                f"cost-of-shopping-{list_item_ids[1]}": "200",
+                f"cost-of-shopping-{list_item_ids[2]}": "300",
+                f"cost-of-other-{list_item_ids[0]}": "10",
+                f"cost-of-other-{list_item_ids[1]}": "20",
+                f"cost-of-other-{list_item_ids[2]}": "30",
+                f"days-a-week-{list_item_ids[0]}": "5",
+                f"days-a-week-{list_item_ids[1]}": "4",
+                f"days-a-week-{list_item_ids[2]}": "3",
+                "based-checkbox-answer": "UK based supermarkets",
+                "extra-static-answer": "5",
+            }
+        )
+        self.post({"extra-spending-answer": "50"})
+        self.post({"extra-spending-method-answer": "Yes"})
+        self.assertInBody(
+            "We calculate the total cost of your weekly shopping to be £715.00. Is this correct?"
+        )
+        self.assertNotInBody("How many days a week do you shop at Tesco?")
+        self.assertInBody("How much do you spend on groceries at Tesco?")
+        self.assertInBody("How much do you spend on other items at Tesco?")
+        self.post()
+        self.assertInBody(
+            "We calculate the total visits to the shop to be 12. Is this correct?"
+        )
+        self.assertNotInBody("How much do you spend on groceries at Tesco?")
+        self.assertNotInBody("How much do you spend on other items at Tesco?")
+        self.assertInBody("How many days a week do you shop at Tesco?")
+        self.post()
+        self.post()
+        self.post()
+        self.post({"weekly-car-trips-answer": "15"})
```

Comment:
Can we add a functional test to show that we use answer from section 1 in section 2, this seems to be testing very general behaviour, why was it done in integration tests?

Reply:
I've added the test you suggested into `test_calculated_summary_context` so that the blocks being filtered out is more explicit than the integration test that got deleted - 100% coverage either way now